### PR TITLE
Architectural cleanup to make cross plat code sharing easier

### PR DESCRIPTION
**Pull request by RPangrle-MS**
Split up ADAuthenticationWebViewController and ADAuthenticationBroker, this should allow us to reuse much more code between iOS and Mac.

Platform specific bits of ADAuthenticationBroker go into ADWindowController, platform specific bits of ADAuthenticationWebViewController go into ADWebViewController. These classes will have different implementations on iOS vs Mac, but they should be more focussed solely on UI API differences.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324%23issuecomment-104788691%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324%23issuecomment-105605156%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324%23discussion_r31058176%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324%23discussion_r31075028%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324%23issuecomment-104788691%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hi%20__%40RPangrle-MS__%2C%20I%27m%20your%20friendly%20neighborhood%20Microsoft%20Open%20Technologies%2C%20Inc.%20Pull%20Request%20Bot%20%28You%20can%20call%20me%20MSOTBOT%29.%20Thanks%20for%20your%20contribution%21%5Cr%5Cn%20%20%20%20%3Cspan%3EYou%27ve%20already%20signed%20the%20contribution%20license%20agreement.%20Thanks%21%3C/span%3E%5Cr%5Cn%20%20%20%20%20%20%20%20%3Cp%3EThe%20agreement%20was%20validated%20by%20Microsoft%20Open%20Technologies%2C%20Inc.%20and%20real%20humans%20are%20currently%20evaluating%20your%20PR.%3C/p%3E%5Cr%5Cn%5Cr%5CnTTYL%2C%20MSOTBOT%3B%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-05-22T22%3A24%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9519461%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/msotclas%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-05-26T17%3A08%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20f07cc2af41dae66832f153b1e3ce0a38c7bbb823%20ADALiOS/ADALiOS/ADAuthenticationBroker.m%20161%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324%23discussion_r31058176%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22The%20interaction%20between%20the%20NTLM%20Session%20Handler%20and%20launching%20the%20window%20is%20not%20as%20clean%20as%20it%20should%20be.%20I%27m%20going%20to%20add%20a%20notification%20here%20to%20better%20scope%20when%20that%20is%20happening.%22%2C%20%22created_at%22%3A%20%222015-05-26T17%3A29%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-05-26T20%3A18%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOS/ADAuthenticationBroker.m%3AL218-231%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324#issuecomment-104788691'>General Comment</a></b>
- <a href='https://github.com/msotclas'></a> Hi __@RPangrle-MS__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
<span>You've already signed the contribution license agreement. Thanks!</span>
<p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>
TTYL, MSOTBOT;
- [x] <a href='#crh-comment-Pull f07cc2af41dae66832f153b1e3ce0a38c7bbb823 ADALiOS/ADALiOS/ADAuthenticationBroker.m 161'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324#discussion_r31058176'>File: ADALiOS/ADALiOS/ADAuthenticationBroker.m:L218-231</a></b>
- <a href='https://github.com/RPangrle-MS'></a> The interaction between the NTLM Session Handler and launching the window is not as clean as it should be. I'm going to add a notification here to better scope when that is happening.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/324?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/324?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/324'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by msotclas** [comment]
Hi __@RPangrle-MS__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>

TTYL, MSOTBOT;

**Comment by RPangrle-MS** [comment]
:+1:

File: ADALiOS/ADALiOS/ADAuthenticationBroker.m [hunk 1]
```diff
@@ -251,14 +218,13 @@ - (void)start:(NSURL *)startURL
...
-        
-        if (parent)
+        _windowController = [[ADWindowController alloc] init];
+        if (_windowController)
         {
+#if TARGET_OS_IPHONE
+            [_windowController setParentController:parent];
+            [_windowController setFullScreen:fullScreen];
+#endif // TARGET_OS_IPHONE
             _ntlmSession = [ADNTLMHandler startWebViewNTLMHandlerWithError:nil];
```
Comment (RPangrle-MS) [inline ADALiOS/ADALiOS/ADAuthenticationBroker.m]: The interaction between the NTLM Session Handler and launching the window is not as clean as it should be. I'm going to add a notification here to better scope when that is happening.

Comment (RPangrle-MS) [inline ADALiOS/ADALiOS/ADAuthenticationBroker.m]: :+1:

**Pull request merged**